core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -253,4 +260,27 @@ private static void deleteFiles(FileIO io, Set<ManifestFile> allManifests) {
           }
         });
   }
+
+  private static String fullTableName(String catalogName, TableIdentifier identifier) {
+    StringBuilder sb = new StringBuilder();
+
+    if (catalogName.contains("/") || catalogName.contains(":")) {
+      // use / for URI-like names: thrift://host:port/db.table
+      sb.append(catalogName);
+      if (!catalogName.endsWith("/")) {
+        sb.append("/");
```

Comment:
nit: it is slightly better to call `append(char)` instead of `append(String)` if appending a single character.